File: src/main/scala/info/crlog/higgs/protocols/boson/json/DemoClient.scala
package info.crlog.higgs.protocols.boson.json

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
object DemoClient {
  def main(args: Array[String]) {
//    val mapper = new ObjectMapper()
//    mapper.registerModule(DefaultScalaModule)
//    println(mapper.
//      writeValueAsString(new Message("test", Some(Seq("d")), Some("callback"), 102)))



        val client = new BosonClient("localhost", 12001)
        client.connect()
        for (i <- 1 to 1) {
          client.invoke("test", Seq(i), (m: Array[String]) => {
            println("received:",m)
          },false)
    //      client.invoke("test", Seq(i), (m: Message) => {
    //        println("msg:",m)
    //      },false)
        }
  }
}

File: higgs-boson/src/main/java/io/higgs/boson/demo/DemoClient.java
package io.higgs.boson.demo;

import com.google.common.base.Optional;
import io.higgs.boson.BosonClient;
import io.higgs.boson.BosonConnectFuture;
import io.higgs.events.listeners.ChannelEventListener;
import io.higgs.functional.Function1;
import io.netty.channel.ChannelHandlerContext;

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
public class DemoClient {
    protected DemoClient() {
    }

    public static void main(String... args) {
        BosonClient client = new BosonClient();
        client.onException(new ChannelEventListener() {
            public void triggered(ChannelHandlerContext ctx, Optional<Throwable> ex) {
                ex.get().printStackTrace();
            }
        });

        BosonConnectFuture future = client.connect("BosonDemo", "localhost", 8080, true);
        for (int i = 0; i < 500000; i++) {
            //even if we're not connected the client will buffer messages until we are, then send them
            future.invoke("polo", new Function1<PoloExample>() {
                public void apply(PoloExample a) {
                    System.out.println(a.i);
                }
            }, new PoloExample(i));
        }
        System.out.println("Done sending");
    }
}
